src/staff_ghosts.c

```diff
@@ -212,261 +212,275 @@ void func_80005310(void) {
+    gControllerSix->rawStickX = stickVal;
 
-    if (temp_v0 < 0x80U) {
-        phi_v1 = (s16) (temp_v0 & 0xFF);
+    if (stickBytes < 0x80U) {
+        stickVal = (s16) (stickBytes & 0xFF);
     } else {
-        phi_v1 = (s16) (temp_v0 | (~0xFF));
+        stickVal = (s16) (stickBytes | (~0xFF));
     }
 
-    gControllerSix->rawStickY = phi_v1;
+    gControllerSix->rawStickY = stickVal;
 
-    if (temp_a0 & 0x80000000) {
-        phi_a2 |= A_BUTTON;
+    if (inputs & 1 << 31) {   //ABZ(start) (D-Up D-Dn D-L D-R) (xx LR) (C-up C-Dn C-L C-R)
+        buttons = A_BUTTON;
     }
-    if (temp_a0 & 0x40000000) {
-        phi_a2 |= B_BUTTON;
+    if (inputs & 1 << 30) {
+        buttons |= B_BUTTON;
     }
-    if (temp_a0 & 0x20000000) {
-        phi_a2 |= Z_TRIG;
+    if (inputs & 1 << 29) {
+        buttons |= Z_TRIG;
     }
-    if (temp_a0 & 0x10000000) {
-        phi_a2 |= R_TRIG;
+    /* This is strange. My assumption is that they remapped the R-Trigger at some point after creating the ghosts.
+    Instead of re-recording the ghosts, I guess they just fixed it here */
+    if (inputs & 1 << 28) {
+        buttons |= R_TRIG;
     }
-    temp_v1 = gControllerSix->buttonPressed & 0x1F0F;
-    gControllerSix->buttonPressed = (phi_a2 & (phi_a2 ^ D_80162D88)) | temp_v1;
+    buttons_temp = gControllerSix->buttonPressed & 0x1F0F;
+    gControllerSix->buttonPressed = (buttons & (buttons ^ sButtonsPrevPlayerGhost)) | buttons_temp;
 
-    temp_v1 = gControllerSix->buttonDepressed & 0x1F0F;
-    gControllerSix->buttonDepressed = (D_80162D88 & (phi_a2 ^ D_80162D88)) | temp_v1;
-    D_80162D88 = phi_a2;
-    gControllerSix->button = phi_a2;
+    buttons_temp = gControllerSix->buttonDepressed & 0x1F0F;
+    gControllerSix->buttonDepressed = (sButtonsPrevPlayerGhost & (buttons ^ sButtonsPrevPlayerGhost)) | buttons_temp;
+    sButtonsPrevPlayerGhost = buttons;
+    gControllerSix->button = buttons;
 
-    if (D_80162D8C == 0) {
-        D_80162D90++;
-        D_80162D8C = (s32) (D_80162D94[D_80162D90] & 0xFF0000);
+    if (sPlayerGhostFramesRemaining == 0) {
+        sPlayerGhostDataIdx++;
+        sPlayerGhostFramesRemaining = (s32) (sPlayerGhostData[sPlayerGhostDataIdx] & 0xFF0000);
     } else {
-        D_80162D8C += (s32) 0xFFFF0000;
+        sPlayerGhostFramesRemaining -= (s32) 0x10000;
     }
 }
 
-void func_8000599C(void) {
-    s16 temp_a2;
-    u32 phi_a3;
-    u32 temp_v1;
-    u32 temp_v2;
-    u32 temp_v0;
-    u32 temp_t0;
-    u32 temp_a0_2;
-
-    if (((D_80162DB8 >= 0x1000) || ((gPlayerOne->unk_0CA & 2) != 0)) || ((gPlayerOne->unk_0CA & 8) != 0)) {
-        D_80162DF8 = 1;
+void save_player_inputs(void) {
+    s16 buttons;
+    u32 inputs;
+    u32 stickX;
+    u32 stickY;
+    u32 inputCounter;
+    u32 prevInputsWCounter;
+    u32 prevInputs;
+    /* Input file is too long or picked up by lakitu or Out of bounds
+    Not sure if there is any way to be considered out of bounds without lakitu getting called */
```

Comment:
iirc, there is some geography that can set oob when driven on. But like you say here, lakitu gets called.